src/main/java/org/broadinstitute/gatk/nativebindings/NativeLibrary.java

```diff
@@ -0,0 +1,21 @@
+package org.broadinstitute.gatk.nativebindings;
+
+import java.nio.file.Path;
+
+/**
+ * Common interface for all native libraries.
+ */
+public interface NativeLibrary {
+    /**
+     * Returns the path to this native library.
+     * Typically this is the path to the shared object file. GATK will use this path to load the native library
+     * using <code>System.load(path.toString())</code>
```

Comment:
Path will likely be location on the classpath, in which case `System.load()` will not be run on this path directly -- the library will have to be extracted to a temp location first before loading. Update docs to clarify.